website/content/docs/platform/k8s/helm/terraform.mdx

```diff
@@ -0,0 +1,273 @@
+  --set='server.ha.enabled=true' \
+  --set='server.ha.raft.enabled=true'
+```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig>
+
+```yaml
+server:
+  ha:
+    enabled: true
+    raft:
+      enabled: true
+```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig>
+
+```hcl
+provider "helm" {
+  kubernetes {
+    config_path = "~/.kube/config"
+  }
+}
+
+resource "helm_release" "vault" {
+  name       = "vault"
+  repository = "https://helm.releases.hashicorp.com"
+  chart      = "vault"
+
+  set {
+    name  = "server.ha.enabled"
+    value = "true"
+  }
+  set {
+    name  = "server.ha.raft.enabled"
+    value = "true"
+  }
+}
+```
+
+</CodeBlockConfig>
+</CodeTabs>
+
+The values file can also be used directly in the Terraform configuration with the [`values` directive](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#values).
+
+## Further Examples
+
+### Vault config as a multi-line string
+
+<CodeTabs>
+<CodeBlockConfig>
+
+```yaml
+server:
+  ha:
+    enabled: true
+    raft:
+      enabled: true
+      setNodeId: true
+      config: |
+        ui = false
+
+        listener "tcp" {
+          tls_disable = 1
+          address = "[::]:8200"
+          cluster_address = "[::]:8201"
+        }
+
+        storage "raft" {
+          path    = "/vault/data"
+        }
+
+        service_registration "kubernetes" {}
+
+        seal "awskms" {
+          region     = "us-west-2"
+          kms_key_id = "alias/my-kms-key"
+        }
+```
+
+</CodeBlockConfig>
+<CodeBlockConfig>
+
+```hcl
+resource "helm_release" "vault" {
+  name       = "vault"
+  repository = "https://helm.releases.hashicorp.com"
+  chart      = "vault"
+
+  set {
+    name  = "server.ha.enabled"
+    value = "true"
+  }
+  set {
+    name  = "server.ha.raft.enabled"
+    value = "true"
+  }
+  set {
+    name  = "server.ha.raft.setNodeId"
+    value = "true"
+  }
+  set {
+    name  = "server.ha.raft.config"
+    value = <<EOT
+ui = false
+
+listener "tcp" {
+  tls_disable = 1
+  address = "[::]:8200"
+  cluster_address = "[::]:8201"
+}
+
+storage "raft" {
+  path    = "/vault/data"
+}
+
+service_registration "kubernetes" {}
+
+seal "awskms" {
+  region     = "us-west-2"
+  kms_key_id = "alias/my-kms-key"
+}
+EOT
+  }
+```
+
+</CodeBlockConfig>
+</CodeTabs>
+
+### Lists of volumes and volumeMounts
+
+<CodeTabs>
+<CodeBlockConfig>
+
+```yaml
+server:
+  volumes:
+    - name: userconfig-my-gcp-iam
+      secret:
+        defaultMode: 420
+        secretName: my-gcp-iam
+
+  volumeMounts:
+    - mountPath: /vault/userconfig/my-gcp-iam
+      name: userconfig-my-gcp-iam
+      readOnly: true
+```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig>
+
+```hcl
+resource "helm_release" "vault" {
+  name       = "vault"
+  repository = "https://helm.releases.hashicorp.com"
+  chart      = "vault"
+
+  set {
+    name  = "server.volumes[0].name
+    value = "userconfig-my-gcp-iam"
+  }
+  set {
+    name  = "server.volumes[0].secret.defaultMode"
+    value = 420
```

Comment:
Elsewhere, we use stringly typed booleans - should this be a string as well, or does it work ok as a number?

Reply:
It seems to work ok as a number. terraform plan shows:

```
      + set {
          + name  = "server.volumes[0].secret.defaultMode"
          + value = "420"
        }
```

I think it's because the helm provider treats it as a string by default, but we can explicitly set it as a string too.